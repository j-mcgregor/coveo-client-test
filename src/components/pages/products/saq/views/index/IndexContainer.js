import React, { useState } from 'react';
import PropTypes from 'prop-types';
import SuggestedList from '../../sections/SuggestedList';
import S from '../../../../../../static/styles';

const ProductsIndex = ({ results, selectProduct, loading, theme }) => {
  const [cols, setCols] = useState(2);
  const palette = theme.theme;

  const renderContent = () => (
    <SuggestedList
      results={results}
      selectProduct={selectProduct}
      title="Results"
      cols={cols}
      truncate
      palette={palette}
    />
  );

  return (
    <div>
      <div className="container grid-lg">
        <S.OptionsPanel className="columns" palette={palette}>
          <div className="column col-6" />
          <div className="column col-6 text-right tile-size">
            <i
              className="fas fa-list"
              onClick={() => setCols(6)}
              role="presentation"
            />
            <i
              className="fas fa-th-large"
              onClick={() => setCols(2)}
              role="presentation"
            />
          </div>
        </S.OptionsPanel>
        {loading ? (
          <S.SpinnerContainer className="columns">
            <div className="loading loading-lg" />
          </S.SpinnerContainer>
        ) : (
          <div className="columns">{renderContent()}</div>
        )}
      </div>
    </div>
  );
};

ProductsIndex.propTypes = {
  selectProduct: PropTypes.func.isRequired,
  results: PropTypes.arrayOf(PropTypes.shape({})),
  loading: PropTypes.bool,
  theme: PropTypes.shape({})
};

ProductsIndex.defaultProps = {
  results: [],
  loading: false,
  theme: {}
};

export default ProductsIndex;
