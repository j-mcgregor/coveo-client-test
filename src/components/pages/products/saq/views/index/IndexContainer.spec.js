/* eslint-disable react/prop-types */
import React from 'react';
import { shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import IndexContainer from './IndexContainer';

describe('IndexContainer', () => {
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      selectProduct: jest.fn(),
      results: [],
      loading: false
    };
  });

  it('should render', () => {
    const wrapper = shallow(
      <ThemeProvider theme={{ light: { color: 'yellow' } }}>
        <IndexContainer {...defaultProps} />
      </ThemeProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
