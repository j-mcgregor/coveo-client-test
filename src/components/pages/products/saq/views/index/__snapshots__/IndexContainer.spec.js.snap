// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexContainer should render 1`] = `ShallowWrapper {}`;
